### PR TITLE
Allow Sending of Email Asynchronously

### PR DESCRIPTION
**Pull request by Henk3000**
When using Send Better Email in a flow the content of the email will not incorporate changes made to objects elsewhere in the flow. This is an issue for example when using the action in a record triggered content where no "wait" or "pause" functions are available to commit the work to the database.

Therefore I added an option to be able to sent the email(s) via a future method.

### CODE REVIEW

File: flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls [hunk 1]
```diff
@@ -154,8 +177,6 @@ public without sharing class SendBetterEmail {
         subject = null;
       }
 
-      SendBetterEmailResponse thisResponse = new SendBetterEmailResponse();
-      thisResponse.index = requestIndex;
```
Comment (Henk3000) [inline flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls:158]: moved creating the response up to generate a response for future requests

File: flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls [hunk 2]
```diff
@@ -70,6 +83,9 @@ public without sharing class SendBetterEmail {
 
     // Collect all inputs for SOQL
     for (SendBetterEmailRequest curRequest : requests) {
+      if (curRequest.sendAsync && !isFuture) {
+        continue;
```
Comment (Henk3000) [inline flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls:87]: no need to query for requests that are not to be processed, if isFuture (being in future context) the request should be processed

File: flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls [hunk 3]
```diff
@@ -123,6 +139,13 @@ public without sharing class SendBetterEmail {
     for (SendBetterEmailRequest curRequest : requests) {
       //Assign Request Number to the Requests
       curRequest.index = requestIndex;
+      SendBetterEmailResponse thisResponse = new SendBetterEmailResponse();
+      thisResponse.index = requestIndex;
+      if (curRequest.sendAsync && !isFuture) {
+        thisResponse.isSuccess = true;
+        responseList.add(thisResponse);
+        continue;
```
Comment (Henk3000) [inline flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls:147]: continue to next request if not in future context (ignore else)

**Comment by alexed1** [comment]
Thanks for this nice enhancement, @Henk3000! Would you be interested in writing a blog post that discusses and demonstrates its use?

**Comment by Henk3000** [comment]
@alexed1 Sure, I will, I'll get on that somewhere this week.

**Comment by alexed1** [comment]
great. let me know when you have it, and I'll merge and package this.

**Comment by alexed1** [comment]
Hi, Henk. Just checking in on this. Do you think you'll have time to create a writeup? I'd love to add this....

**Review by alexed1 — APPROVED** [review]

**Comment by alexed1** [comment]
@jdpond note that this is merged but was never packaged, so it will go into the next patch.

**Comment by jdpond** [comment]
I've tested it by hand - but haven't added it to test class or Send_Better_Email_Testflow flow.  Probably should do this, but prolly not necessary for next patch - just better to have to avoid future unintentional maintenance factors.

**Pull request merged**